### PR TITLE
fix: fix illegal default values in bounded hyperparameters

### LINKED ISSUE #310: NormalIntegerHyperparameter and NormalFloatHyperparameter accepts illegal default values
```
from ConfigSpace import Normal, Integer, Float
int_hyper = Integer("p", bounds=(0.0, 2.0), default=5., distribution=Normal(1.0, 0.2))
float_hyper =  Float("p", bounds=(0.0, 2.0), default=5., distribution=Normal(1.0, 0))
```
Both these lines should throw exceptions since the default values are outside the bounds, but the code runs just fine.

### PR DESCRIPTION
**Pull request by Neonkraft**
Resolves #310

### CODE REVIEW

File: ConfigSpace/hyperparameters/normal_float.pyx [hunk 1]
```diff
@@ -207,7 +207,9 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
                                            q=q_int, log=self.log)
 
     def is_legal(self, value: Union[float]) -> bool:
-        return isinstance(value, float) or isinstance(value, int)
+        return (isinstance(value, float) or isinstance(value, int))  and \
```
Comment (eddiebergman) [inline ConfigSpace/hyperparameters/normal_float.pyx:210]: Can we make this `isinstance(value, (float, int, np.number))`?

File: ConfigSpace/hyperparameters/normal_integer.pyx [hunk 2]
```diff
@@ -195,7 +195,9 @@ cdef class NormalIntegerHyperparameter(IntegerHyperparameter):
                                             q=self.q, log=self.log, meta=self.meta)
 
     def is_legal(self, value: int) -> bool:
-        return isinstance(value, (int, np.int32, np.int64))
+        return (isinstance(value, (int, np.int32, np.int64))) and \
```
Comment (eddiebergman) [inline ConfigSpace/hyperparameters/normal_integer.pyx:198]: Likewise for here `isinstance(value, (int, np.integer))`

**Review by eddiebergman — CHANGES_REQUESTED** [review]

File: ConfigSpace/hyperparameters/normal_float.pyx (hunk 1 above)
Comment (Neonkraft) [reply ConfigSpace/hyperparameters/normal_float.pyx]: Done.

File: ConfigSpace/hyperparameters/normal_integer.pyx (hunk 2 above)
Comment (Neonkraft) [reply ConfigSpace/hyperparameters/normal_integer.pyx]: Done.

**Review by eddiebergman — APPROVED** [review]

**Pull request merged**